js/ConfidenceSliderModel.js

```diff
@@ -111,6 +115,19 @@ export default class ConfidenceSliderModel extends SliderModel {
     return feedbackString;
   }
 
+  onLinkedSubmittedChanged(linkedModel) {
+    this.set('body', this.get('originalBody'));
+    this.set('_isEnabled', (linkedModel.get('_isSubmitted') === true));
+    this.set('_linkedModelSelectedIndex', linkedModel.get('_selectedItem').index);
+    this.checkCanSubmit();
+  }
+
+  _listenToLinkedModel() {
+    this.listenTo(this.linkedModel, {
+      'change:_isSubmitted': this.onLinkedSubmittedChanged
+    });
+  }
```

Comment:
Line 119: `this.set('body', this.get('originalBody'));` 
This line assumes that the model is submitted and sets `body = originalBody`, you can see this in contrast to line 152 `this.set('body', this.get('disabledBody'));` which sets to `disabledBody` if not submitted .

Line 120: `(linkedModel.get('_isSubmitted') === true)`
This checks if the model is submitted or not and sets `_isEnabled` accordingly, as does line 151 with `if (!this.linkedModel.get('_isSubmitted')) { this.set('_isEnabled', false); }`

Line 154 and Line 121: `this.set('_linkedModelSelectedIndex', linkedModel.get('_selectedItem').index);`
These are identical.


```suggestion
  updateFromLinkedModel() {
    const isSubmitted = this.linkedModel.get('_isSubmitted');
    this.set('body', isSubmitted ? this.get('originalBody') : this.get('disabledBody'));
    this.set('_isEnabled', isSubmitted);
    if (isSubmitted) {
      this.set('_linkedModelSelectedIndex', this.linkedModel.get('_selectedItem').index);
    }
    this.checkCanSubmit();
  }

  _listenToLinkedModel() {
    this.listenTo(this.linkedModel, 'change:_isSubmitted', this.updateFromLinkedModel);
  }
```